plugins/module_utils/network/sonic/config/mclag/mclag.py

```diff
@@ -193,6 +200,64 @@ def _state_deleted(self, want, have):
                     commands = update_states(diff_want, "deleted")
         return commands, requests
 
+    def _state_replaced(self, want, have, diff):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        requests = []
+        commands = []
+
+        diff1 = get_diff(have, want, TEST_KEYS, is_skeleton=True)
+        if diff1:
+            requests = self.get_delete_all_mclag_domain_request(have)
```

Comment:
Only the configuration sections with changes should be deleted.

Reply:
Delete all handled in order to address the change in domain id. 

If the existing configuration is 

mclag domain 100
 source-ip 3.3.3.3
 keepalive-interval 4
 session-timeout 50
 delay-restore 300

if the user wanted to replace the domain id alone, 

mclag domain 10 <<<<
 source-ip 3.3.3.3
 keepalive-interval 4
 session-timeout 50
 delay-restore 300

Delete all and re-calculation of facts required to handle this scenario.